src/forms/MatcherFields.tsx

```diff
@@ -101,14 +101,29 @@ function ItemTemplate(file: { name: string }) {
 
 export function FileUploader() {
   const header = (options: FileUploadHeaderTemplateOptions) => <ButtonGroup m={2}>{options.chooseButton}</ButtonGroup>
+
   const empty = () =>
       <VStack flexGrow={1}>
         <Icon boxSize='30%' as={AiOutlineCloudUpload} color='gray.200' />
         <Text color='gray.600'>drag & drop a file here</Text>
       </VStack>
-  return <VStack as={ChakraFileUpload} border='2px dashed' borderColor='gray.200' flexDir='column-reverse'
-                 justify='center' m={8} url='./upload' chooseLabel='Select file' rounded='3xl'
-                 itemTemplate={ItemTemplate} headerTemplate={header} emptyTemplate={empty} />
+
+     /*<VStack as={ChakraFileUpload} border='2px dashed' borderColor='gray.200' flexDir='column-reverse'
+                   justify='center' m={8} url='./upload' chooseLabel='Select file' rounded='3xl'
+itemTemplate={ItemTemplate} headerTemplate={header} emptyTemplate={empty} />*/
+      return <FileUpload name="upload" url='./upload' 
+      onSelect={(event)=>{ 
+                  const fileType = event.files[0].type
+                    if(fileType === 'text/plain' || fileType === 'text/csv' || fileType === 'application/json'){ 
+                      const fileContent = event.files[0].text();
+                      fileContent.then(result => {
+                        const emailList = result.match(/([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9._-]+)/gi)
```

Comment:
Since we already use a form validation schema (Yup) and this regex string triggers a security issue with sonarcloud anyway I would suggest using the schema instead of defining an email regex separately.

This is the general setup including all the formatting:
```
export function FileUploader() {
  const allowedTypes = ['text/plain', 'text/csv', 'application/json']

  const header = (options: FileUploadHeaderTemplateOptions) =>
      <ButtonGroup w='full' justifyContent='center'>
        {options.chooseButton}
      </ButtonGroup>

  const empty =
      <VStack h='full'>
        <Icon boxSize='30%' as={AiOutlineCloudUpload} color='gray.200' />
        <Text color='gray.600'>drag & drop a file here</Text>
      </VStack>

  const itemTemplate = (file: any) =>
      <VStack h='full' fontSize='sm' spacing={4}>
        <Icon boxSize='4rem' opacity={0.1} as={ImFileText2}/>
        <Text textAlign='start' fontWeight={600}>{file.name}</Text>
      </VStack>

  const parseEmails = (content: string, fileType: string) => {
    return fileType === 'application/json' ? JSON.parse(content)['students'] : content.split(/[\r\n]+/)
  }

  return (
      <Field name='students' children={(fieldProps: FieldProps) =>
          <FormControl isInvalid={fieldProps.meta.value && fieldProps.meta.error}>
            <ChakraFileUpload p={4} minH='2xs' headerTemplate={header} emptyTemplate={empty} itemTemplate={itemTemplate}
                              chooseLabel='Select File' rounded='3xl' border='2px dashed' borderColor='gray.200'
                              onSelect={event => {
                                const fileType = event.files[0]?.type
                                if (allowedTypes.includes(fileType))
                                    event.files[0].text().then(content =>
                                        fieldProps.form.setFieldValue('students', parseEmails(content, fileType)))
                                else fieldProps.form.setFieldError('students', 'File type is not supported.') }}
                              progressBarTemplate={
              <Center textAlign='center' fontWeight={600} color='purple.500'>
                <Text maxW='xs'>{fieldProps.meta.error || `Detected ${fieldProps.field.value.length} emails.`}</Text>
              </Center>} />
          </FormControl>} />
  )
}
```

and then change the baseSchema (in Schemas.tsx) on line 42-43:

```
  students: array().default([]).of(string().ensure().email(context =>
      <span key={context.path}>Line {Number(context.path?.match(/\d+/g))+1} is not a valid email address<br/></span>)),
```

This setup only supports whatever parsing we put inside the function parseEmails. For now we support:

- A json file with a field called 'students' and then a list of string values (so like this: { 'students': ['...@email.com', '...', ...] }
- A .txt or .csv file (we try to parse those by separating the string in multiple ways, specifically by new line / by comma / by semicolon)

If there are other file types / parsing methods you want to add then feel free to expand the parseEmails function.